librz/debug/p/native/bt/windows-x64.c

```diff
@@ -5,6 +5,207 @@
 #include <mdmp_windefs.h>
 #include <librz/bin/format/pe/pe64.h>
 
+// the original PE64_UNWIND_CODE is a union
+typedef struct windows_x64_unwind_code_t {
+	ut8 CodeOffset;
+	ut8 UnwindOp; // : 4;
+	ut8 OpInfo; // : 4;
+	ut16 FrameOffset;
+	ut64 offset;
+} Win64UnwindCode;
+
+typedef struct windows_x64_unwind_info_t {
+	ut8 Version; // : 3;
+	ut8 Flags; // : 5;
+	ut8 SizeOfProlog;
+	ut8 CountOfCodes;
+	ut8 FrameRegister; // : 4;
+	ut8 FrameOffset; // : 4;
+	Win64UnwindCode *code;
+} Win64UnwindInfo;
+
+enum {
+	AMD64_INDEX_RAX = 0,
+	AMD64_INDEX_RCX,
+	AMD64_INDEX_RDX,
+	AMD64_INDEX_RBX,
+	AMD64_INDEX_RSP,
+	AMD64_INDEX_RBP,
+	AMD64_INDEX_RSI,
+	AMD64_INDEX_RDI,
+	AMD64_INDEX_R8,
+	AMD64_INDEX_R9,
+	AMD64_INDEX_R10,
+	AMD64_INDEX_R11,
+	AMD64_INDEX_R12,
+	AMD64_INDEX_R13,
+	AMD64_INDEX_R14,
+	AMD64_INDEX_R15,
+	AMD64_INDEX_RIP,
+};
+
+static Win64UnwindCode *windows_x64_unwind_code_new(RzDebug *dbg, ut64 at, ut32 count) {
+	ut8 code[sizeof(ut16)] = { 0 };
+
+	Win64UnwindCode *uc = RZ_NEWS0(Win64UnwindCode, count);
+	if (!uc) {
+		return NULL;
+	}
+
+	for (ut32 i = 0; i < count; ++i, at += sizeof(ut16)) {
+		if (!dbg->iob.read_at(dbg->iob.io, at, code, sizeof(code))) {
+			free(uc);
+			return NULL;
+		}
+		uc[i].CodeOffset = code[0];
+		uc[i].UnwindOp = code[1] & 0x0F;
+		uc[i].OpInfo = code[1] >> 4;
+		uc[i].FrameOffset = rz_read_le16(code);
+		uc[i].offset = at;
+	}
+
+	return uc;
+}
+
+static Win64UnwindInfo *windows_x64_unwind_info_new(RzDebug *dbg, ut64 at) {
```

Comment:
I wonder if it's possibleble to get rid of this code duplication

Reply:
it's not really code duplication since the native debugger needs the UnwindCode data, meanwhile the other one does not. but i see your point